provisioning/service/src/Enrollments/TpmAttestation.cs

```diff
@@ -25,15 +23,15 @@ public sealed class TpmAttestation : Attestation
         /// with both endorsement and storage root keys. Only the endorsement
         /// key is mandatory.
         /// </remarks>
-        /// <param name="endorsementKey">the <c>string</c> with the TPM endorsement key. It cannot be <c>null</c> or empty.</param>
-        /// <param name="storageRootKey">the <c>string</c> with the TPM storage root key. It can be <c>null</c> or empty.</param>
-        /// <exception cref="ArgumentException">if the endorsementKey is <c>null</c> or empty.</exception>
+        /// <param name="endorsementKey">the string with the TPM endorsement key. It cannot be null or empty.</param>
+        /// <param name="storageRootKey">the string with the TPM storage root key. It can be null or empty.</param>
+        /// <exception cref="ArgumentException">if the endorsementKey is null or empty.</exception>
         [JsonConstructor]
         public TpmAttestation(string endorsementKey, string storageRootKey = null)
         {
             try
             {
-                EndorsementKey = endorsementKey;
+                EndorsementKey = endorsementKey ?? throw new ArgumentNullException(nameof(endorsementKey));
```

Comment:
If it can't be empty, a null check (??) won't cut it.

How about:

```suggestion
                Argument.AssertNotNullOrEmpty(endorsementKey, nameof(endorsementKey));
                EndorsementKey = endorsementKey;
```